crypto/fipsmodule/ec/p384.c

```diff
@@ -0,0 +1,524 @@
+  fiat_p384_sub(x_out, x_out, j);
+  fiat_p384_sub(x_out, x_out, v);
+  fiat_p384_sub(x_out, x_out, v);
+
+  // y_out = r(V-x_out) - 2 * s1 * J
+  fiat_p384_sub(y_out, v, x_out);
+  fiat_p384_mul(y_out, y_out, r);
+  fiat_p384_felem s1j;
+  fiat_p384_mul(s1j, s1, j);
+  fiat_p384_sub(y_out, y_out, s1j);
+  fiat_p384_sub(y_out, y_out, s1j);
+
+  fiat_p384_cmovznz(x_out, z1nz, x2, x_out);
+  fiat_p384_cmovznz(x3, z2nz, x1, x_out);
+  fiat_p384_cmovznz(y_out, z1nz, y2, y_out);
+  fiat_p384_cmovznz(y3, z2nz, y1, y_out);
+  fiat_p384_cmovznz(z_out, z1nz, z2, z_out);
+  fiat_p384_cmovznz(z3, z2nz, z1, z_out);
+}
+
+// OPENSSL EC_METHOD FUNCTIONS
+
+// Takes the Jacobian coordinates (X, Y, Z) of a point and returns (X', Y') =
+// (X/Z^2, Y/Z^3).
+static int ec_GFp_nistp384_point_get_affine_coordinates(
+    const EC_GROUP *group, const EC_RAW_POINT *point, EC_FELEM *x_out,
+    EC_FELEM *y_out) {
+  if (ec_GFp_simple_is_at_infinity(group, point)) {
+    OPENSSL_PUT_ERROR(EC, EC_R_POINT_AT_INFINITY);
+    return 0;
+  }
+
+  fiat_p384_felem z1, z2;
+  fiat_p384_from_generic(z1, &point->Z);
+  fiat_p384_inv_square(z2, z1);
+
+  if (x_out != NULL) {
+    fiat_p384_felem x;
+    fiat_p384_from_generic(x, &point->X);
+    fiat_p384_mul(x, x, z2);
+    fiat_p384_to_generic(x_out, x);
+  }
+
+  if (y_out != NULL) {
+    fiat_p384_felem y;
+    fiat_p384_from_generic(y, &point->Y);
+    fiat_p384_square(z2, z2);  // z^-4
+    fiat_p384_mul(y, y, z1);   // y * z
+    fiat_p384_mul(y, y, z2);   // y * z^-3
+    fiat_p384_to_generic(y_out, y);
+  }
+
+  return 1;
+}
+
+static void ec_GFp_nistp384_add(const EC_GROUP *group, EC_RAW_POINT *r,
+                                const EC_RAW_POINT *a, const EC_RAW_POINT *b) {
+  fiat_p384_felem x1, y1, z1, x2, y2, z2;
+  fiat_p384_from_generic(x1, &a->X);
+  fiat_p384_from_generic(y1, &a->Y);
+  fiat_p384_from_generic(z1, &a->Z);
+  fiat_p384_from_generic(x2, &b->X);
+  fiat_p384_from_generic(y2, &b->Y);
+  fiat_p384_from_generic(z2, &b->Z);
+  fiat_p384_point_add(x1, y1, z1, x1, y1, z1, 0 /* both Jacobian */, x2, y2, z2);
+  fiat_p384_to_generic(&r->X, x1);
+  fiat_p384_to_generic(&r->Y, y1);
+  fiat_p384_to_generic(&r->Z, z1);
+}
+
+static void ec_GFp_nistp384_dbl(const EC_GROUP *group, EC_RAW_POINT *r,
+                                const EC_RAW_POINT *a) {
+  fiat_p384_felem x, y, z;
+  fiat_p384_from_generic(x, &a->X);
+  fiat_p384_from_generic(y, &a->Y);
+  fiat_p384_from_generic(z, &a->Z);
+  fiat_p384_point_double(x, y, z, x, y, z);
+  fiat_p384_to_generic(&r->X, x);
+  fiat_p384_to_generic(&r->Y, y);
+  fiat_p384_to_generic(&r->Z, z);
+}
+
+static void ec_GFp_nistp384_mont_felem_to_bytes(const EC_GROUP *group, uint8_t *out,
+                                         size_t *out_len, const EC_FELEM *in) {
+  size_t len = BN_num_bytes(&group->field);
+  EC_FELEM felem_tmp;
+  fiat_p384_felem tmp;
+  fiat_p384_from_generic(tmp, in);
+  fiat_p384_from_montgomery(felem_tmp.words, tmp);
```

Comment:
Looks like this function call makes the Windows build fail because of uint32_t to uint64_t pointer conversion. It's probably some flag that's not set correctly, could you please check that?